src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -759,8 +759,17 @@ private void Parse()
                 @"((?<username>[^:@]+)(:(?<password>[^:@]*))?@)?" +
                 serversPattern + @"(/" + databasePattern + ")?/?" + optionsPattern + "$";
 
+            Match badPercentMatch = null;
+            var hexDigit= "0-9A-Fa-f";
+            var percentPattern = $"(%[^{hexDigit}])|"  // % followed by a non-hexadecimal digit OR
+                                 +$"(%[{hexDigit}][^{hexDigit}])"; // % followed by a hexadecimal digit and a non-hexadecimal digit
+            if (_originalConnectionString.Contains("%"))
```

Comment:
I think this line was previously necessary when we were exploring the other implementation option, but is it still necessary with the current approach?

Reply:
Not necessary, but a possible performance micro-optimization if connection strings hardly ever contain % escape sequences because this check would be faster than running an expensive regular expression check.

Reply:
Sgtm! Resolving !